src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
+
+  const getCellValue = (row, index) => {
+    if (row.children[index].innerText !== undefined) {
+      return row.children[index].innerText;
+    } else {
+      return row.children[index].textContent;
+    }
+  };
+
+  const sortFilter = (index, order) => (rowA, rowB) => {
+    const valueA = getCellValue(rowA, index);
+    const valueB = getCellValue(rowB, index);
+
+    let numberA;
+    let numberB;
+
+    if (valueA.charCodeAt(0) === 36) {
+      const arrA = Array.from(valueA);
+      const arrB = Array.from(valueB);
+
+      arrA.shift();
+      arrB.shift();
+
+      numberA = arrA.join('');
+      numberB = arrB.join('');
+    } else {
+      numberA = valueA;
+      numberB = valueB;
+    }
+
+    const numA = parseFloat(numberA);
+    const numB = parseFloat(numberB);
+
+    if (order === 'ASC') {
+      if (!isNaN(numA) && !isNaN(numB)) {
+        return numA - numB;
+      }
+
+      return valueA.localeCompare(valueB);
+    } else {
+      if (!isNaN(numA) && !isNaN(numB)) {
+        return numB - numA;
+      }
+
+      return valueB.localeCompare(valueA);
+    }
+  };
+
+  document.querySelectorAll('th').forEach((header, index) => {
+    let sortOrder = null;
+
+    header.addEventListener('click', () => {
+      // const table = header.closest('table');
+      const tbody = header.closest('table').querySelector('tbody');
+      // const table = header.closest('table');
+      // const tbody = table.querySelector('tbody');
+      const rowsArr = Array.from(tbody.querySelectorAll('tr'));
+
+      if (sortOrder === 'ASC') {
+        sortOrder = 'DESC';
+      } else {
+        sortOrder = 'ASC';
+      }
+
+      rowsArr.sort(sortFilter(index, sortOrder));
+
+      tbody.append(...rowsArr);
+    });
+  });
+
+  document.querySelectorAll('tbody tr').forEach((row) => {
+    row.addEventListener('click', () => {
+      document.querySelectorAll('tbody tr').forEach((r) => {
+        r.classList.remove('active');
+        r.style.backgroundColor = '';
+        r.style.color = '';
+      });
+
+      row.classList.add('active');
+      row.style.backgroundColor = 'rgb(208, 208, 208)';
+      row.style.color = 'black';
+    });
+  });
+
+  const form = document.createElement('form');
+
+  form.classList.add('new-employee-form');
+
+  const nameLabel = document.createElement('label');
+  const nameInput = document.createElement('input');
+
+  nameLabel.textContent = 'Name:';
+  nameInput.type = 'text';
+  nameInput.name = 'name';
+  nameInput.dataset.qa = 'name';
+  nameInput.required = true;
+  nameInput.setAttribute('data-qa', 'name');
+  nameLabel.appendChild(nameInput);
+
+  const positionLabel = document.createElement('label');
+  const positionInput = document.createElement('input');
+
+  positionLabel.textContent = 'Position:';
+  positionInput.type = 'text';
+  positionInput.name = 'position';
+  positionInput.dataset.qa = 'position';
+  positionInput.required = true;
+  positionInput.setAttribute('data-qa', 'position');
```

Comment:
Redundant code: The `setAttribute` method is used to set the `data-qa` attribute, which is already set using the `dataset` property. Consider removing one of these to avoid redundancy.